### ISSUE TITLE
Invalid book tag in confirmtransaction in 1.15

### ISSUE THREAD

**Issue by Bjoern-B** [MLG-Fortress/PrettySimpleShop#28]
I get this *Invalid book tag*
![Unbenannt](https://user-images.githubusercontent.com/38884487/71028931-8ef68200-210e-11ea-8b73-63614337d44b.PNG)

if I have "confirmTransactions: true", im using the latest version 1.7.1 if I then try to buy the item again the book does not show and I get the items.

Thanks in advance.

**Comment by RoboMWM** [comment]
Wonder if 1.15 changed anything. Can you confirm if this is the case? Please also provide output of `/version`

**Comment by Bjoern-B** [comment]
Oh year totaly forgot to mention that im on 1.15
![Unbenannt1](https://user-images.githubusercontent.com/38884487/71065736-1cb78900-2172-11ea-9ea5-68c81d81f8bd.PNG)

just updated to 1.15.1 real quick the problem persists there

![Unbenannt](https://user-images.githubusercontent.com/38884487/71065743-20e3a680-2172-11ea-87ba-252f2e4e4d95.PNG)

**Comment by RoboMWM** [comment]
Ok thanks

If you could find out what changed with books that'd be very helpful

**Comment by Keks4Gaming** [comment]
https://www.spigotmc.org/threads/1-15-invalid-book-tag.408957/

I found this. Maybe its Helpful.

Edit: I scroll through your source and cant find that you title that written book, as it has sayed in the discussion from the link above.

**Comment by RoboMWM** [comment]
Hmmm. So books require a title?

**Comment by Keks4Gaming** [comment]
Since 1.15 books require a title, yes.

**Comment by RoboMWM** [comment]
Mk. I can't seem to login to test server so I haven't been able to tests this, but I'll go ahead and do that if that'll fix it.

**Comment by Keks4Gaming** [comment]
I have 3 server running. 1.14 1.12 and 1.15. As I mentioned in the duplication glitch post, I can test your plugin. 😁

**Comment by RoboMWM** [comment]
Sometimes seeing is believing, especially when the javadocs don't document this requirement.

**Comment by RoboMWM** [comment]
Sometimes seeing is believing, especially when the javadocs don't document this requirement, and in fact explicitly state the opposite:

```java
    /**
     * Checks for the existence of a title in the book.
     *
     * @return true if the book has a title
     */
    boolean hasTitle();

    /**
     * Gets the title of the book.
     * <p>
     * Plugins should check that hasTitle() returns true before calling this
     * method.
     *
     * @return the title of the book
     */
    @Nullable
    String getTitle();
```

**Comment by RoboMWM** [comment]
@Keks4Gaming then I guess see if this fixes it https://ci.appveyor.com/project/RoboMWM39862/prettysimpleshop/builds/31511965/artifacts thanks

**Comment by Keks4Gaming** [comment]
![image](https://user-images.githubusercontent.com/53128111/76983904-6c0e8080-693e-11ea-9711-8908f29c6186.png)

Iam sorry, but it still do the same on 1.15.1. Can i test something for you to find the problem?

**Comment by RoboMWM** [comment]
Looks like I have to set all metadata, not just title. Javadoc really needs to be updated...

**Comment by RoboMWM** [comment]
@Keks4Gaming try https://ci.appveyor.com/project/RoboMWM39862/prettysimpleshop/builds/31566280/artifacts

**Comment by Keks4Gaming** [comment]
![image](https://user-images.githubusercontent.com/53128111/77157096-f7a02280-6aa0-11ea-98a9-d2a5658162a2.png)

Works! But when i click done, left click the chest again and put a number in chat it instant buys without the book.

**Comment by RoboMWM** [comment]
Don't think it's possible to tell when book is closed. So when you click the forbidden button instead it's still in the verify buy state. The confirm button just re-executes the command so that's why it'll work instantly. If you put in a different quantity it'll ask again though.

Tl;Dr working as expected
